src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -199,10 +200,52 @@ class C
 {
     public async void M()
     {
-        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        using (FileStream s = File.Open(""file.txt"", FileMode.Open))
         {
-            byte[] buffer = new byte[fs.Length];
-            await fs.ReadAsync(buffer, 0, (int)fs.Length).ContinueWith(c => {}).ContinueWith(c => {}).ConfigureAwait(false);
+            byte[] buffer = new byte[s.Length];
+            await s.ReadAsync(buffer, 0, (int)s.Length).ContinueWith(c => {}).ContinueWith(c => {}).ConfigureAwait(false);
+        }
+    }
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_Analyzer_NoDiagnostic_AutoCastedToMemory()
+        {
+            return AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream s = File.Open(""path.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[s.Length];
+            await s.ReadAsync(buffer);
```

Comment:
Added these extra new unit tests: If the user calls `ReadAsync` with only a buffer (no offset and count), it gets automatically casted to a `Memory` object. Same with `WriteAsync` -> `ReadOnlyMemory`.